bin/shodan

```diff
@@ -1126,7 +1149,17 @@ def stream(color, fields, separator, limit, datadir, ports, quiet, timeout, stre
         except:
             # For other errors lets just wait a few seconds and try to reconnect again
             # XXX FIXME: this does not seem to reconnect since it's in the while not quit loop!
-            time.sleep(2)
+            quit = True
+            #time.sleep(2)
+    if pipe:
+        try:
+            os.write(pipeout, '{ "_file_closed_at": "%s" } ]' %cur_time )
+        except IOError, e:
+            if e.errno == errno.EPIPE:
+                 pipeout = os.open(pipe, os.O_WRONLY) # simply re-open the pipe 
+            else:
+                print ("got an IOError: %s" %e)
```

Comment:
Use click.ClickException to return errors and the .format() method for creating strings ala:

raise click.ClickException('Pipe received an IO Error: {}'.format(e))